miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -341,6 +344,81 @@ export.apply_storage_offset
     # => [offset_slot_index]
 end
 
+#! Validates all account procedures storage offsets
+#!
+#! Notes:
+#! - For faucets checks that no storage offset is 0 (preventing reserved storage slot access)
+#!
+#! Stack: []
+#! Output: []
+export.validate_storage_offsets
+    # get number of account procedures
+    exec.memory::get_num_procedures
+    # => [num_account_procedures]
+
+    # prepare stack for looping
+    push.0.1
+    # => [start_loop, index, num_account_procedures]
+
+    # check if the account is a faucet
+    exec.get_id exec.is_faucet
+    # => [is_faucet, start_loop, index, num_account_procedures]
+
+    # we do not check if num_account_procedures == 0 here because a valid
+    # account has between 1 and 256 procedures with associated offsets
+    if.true
+        while.true
+            # get storage offset from memory
+            dup exec.get_procedure_storage_offset
+            # => [storage_offset, index, num_account_procedures]
+
+            # assert that storage offset is not 0
+            dup push.0 neq assert.err=ERR_INVALID_FAUCET_STORAGE_OFFSET
+            # => [storage_offset, index, num_account_procedures]
+
+            # TODO: This is a temporary check for faucets. We add 1 to all faucet offsets
+            # to prevent access to the reserved faucet data slot. When the assembler
+            # will support storage offsets remove this check.
+            # check if the storage offset and the number of storage slots are 1
+            dup exec.memory::get_num_storage_slots eq.1 swap eq.1 and
+            # => [not_both_1, storage_offset, index, num_account_procedures]
```

Comment:
For faucet accounts we need to check that:
1. no procedure has offset 0
2. All offsets are within bounds

But considering that the way to check if an offset is within bounds is to take the number of initialised storage slots `num_storage_slots` and checking if it is strictly greater than the offset of each procedure we end up having an edge case for faucets.

If a faucet has only 1 initialised storage slot (being the reserved data slot) than, the offset of all procedures is `1` (since this PR) and the `num_storage_slot` is 1. Hence the bounds check errors out, even though it shouldn't. 

This additional temporary check enables us to not error out on this edge case.

Reply:
> no procedure has offset 0

Ah yes! correct.

> If a faucet has only 1 initialised storage slot (being the reserved data slot) than, the offset of all procedures is `1` (since this PR) and the `num_storage_slot` is 1. Hence the bounds check errors out, even though it shouldn't.

I see. The issue here is not so much related to faucets but that we don't have a way to identify procedures which don't need storage access at all. Let's keep this code for now and fix this as a part of https://github.com/0xPolygonMiden/miden-base/issues/866.